src/control/controlobject.h

```diff
@@ -2,190 +2,110 @@
-        }
-    }
-
-    // Forces the control to 'value', regardless of whether it has a change
-    // request handler attached (identical to setAndConfirm).
-    inline void forceSet(double value) {
-        setAndConfirm(value);
-    }
-
-    // Instantly sets the value of the ControlObject
+    void set(double value);
+    void setAndConfirm(double value);
+    void forceSet(double value);
     static void set(const ConfigKey& key, const double& value);
 
-    // Sets the default value
-    inline void reset() {
-        if (m_pControl) {
-            m_pControl->reset();
-        }
-    }
-
-    inline void setDefaultValue(double dValue) {
-        if (m_pControl) {
-            m_pControl->setDefaultValue(dValue);
-        }
-    }
-    inline double defaultValue() const {
-        return m_pControl ? m_pControl->defaultValue() : 0.0;
-    }
+    void reset();
+    void setDefaultValue(double dValue);
+    double defaultValue() const { return m_defaultValue; }
 
-    // Returns the parameterized value of the object. Thread safe, non-blocking.
     virtual double getParameter() const;
-
-    // Returns the parameterized value of the object. Thread safe, non-blocking.
     virtual double getParameterForValue(double value) const;
-
-    // Returns the parameterized value of the object. Thread safe, non-blocking.
     virtual double getParameterForMidi(double midiValue) const;
-
-    // Sets the control parameterized value to v. Thread safe, non-blocking.
     virtual void setParameter(double v);
+    virtual void setParameterFrom(double v, QObject* pSender = nullptr);
 
-    // Sets the control parameterized value to v. Thread safe, non-blocking.
-    virtual void setParameterFrom(double v, QObject* pSender = NULL);
-
-    // Connects a Qt slot to a signal that is delivered when a new value change
-    // request arrives for this control.
-    // Qt::AutoConnection: Qt ensures that the signal slot is called from the
-    // thread where the receiving object was created
-    // You need to use Qt::DirectConnection for the engine objects, since the
-    // audio thread has no Qt event queue. But be a ware of race conditions in this case.
-    // ref: http://qt-project.org/doc/qt-4.8/qt.html#ConnectionType-enum
     template <typename Receiver, typename Slot>
     bool connectValueChangeRequest(Receiver receiver, Slot func,
                                    Qt::ConnectionType type = Qt::AutoConnection) {
-        bool ret = false;
-        if (m_pControl) {
-          ret = m_pControl->connectValueChangeRequest(receiver, func, type);
-        }
-        return ret;
+        return m_pControl ? m_pControl->connectValueChangeRequest(receiver, func, type) : false;
     }
 
-    // Installs a value-change request handler that ignores all sets.
     void setReadOnly();
 
-  signals:
-    void valueChanged(double);
+    void triggerCallback() {
+        if (m_callback) {
+            m_callback();
+        }
+    }
```

Comment:
you also implement this in the CPP. This results in a compile error (see "one definition rule").